src/lib/homestore_backend/snapshot_receive_handler.cpp

```diff
@@ -120,17 +118,18 @@ int HSHomeObject::SnapshotReceiveHandler::process_blobs_snapshot_data(ResyncBlob
         }
 
         auto blob_data = blob->data()->Data();
-        auto header = r_cast< BlobHeader const* >(blob_data);
-        if (header->valid()) {
-            LOGE("Invalid header found for blob_id {}: [header={}]", blob->blob_id(), header->to_string());
-            return INVALID_BLOB_HEADER;
-        }
-        std::string user_key = header->user_key_size
-            ? std::string(r_cast< const char* >(blob_data + sizeof(BlobHeader)), header->user_key_size)
-            : std::string{};
 
         // Check integrity of normal blobs
         if (blob->state() != static_cast< uint8_t >(ResyncBlobState::CORRUPTED)) {
+            auto header = r_cast< BlobHeader const* >(blob_data);
+            if (!header->valid()) {
```

Comment:
Please also check if `blob state != ResyncBlobState::CORRUPTED`. Because if the header is corrupted on the leader side, leader will treat it as corrupted and send it as it, for this case, just save it and correct later.